src/coreclr/jit/importervectorization.cpp

```diff
@@ -488,9 +503,24 @@ GenTree* Compiler::impExpandHalfConstEquals(GenTreeLclVarCommon* data,
         }
         assert(indirCmp->TypeIs(TYP_INT, TYP_UBYTE));
 
+        if (kind == EndsWith)
+        {
+            // len is expected to be small, so no overflow is possible
+            assert((len * 2) > len);
```

Comment:
Stuff like this gets optimized out since signed overflow is UB.

```suggestion
            assert(!CheckedOps::MulOverflows(len, 2, /* unsignedMul */ false));
```

Reply:
Nit: `/* unsignedMul */ false` -> `CheckedOps::Signed`.

Reply:
Thanks! TIL